operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/ResourceController.java

```diff
@@ -3,53 +3,47 @@
 import io.fabric8.kubernetes.client.CustomResource;
 import io.javaoperatorsdk.operator.processing.event.EventSourceManager;
 
+/**
+ * Interface to be implemented by user-created resource controller classes that operate user-defined
+ * custom resource objects.
+ *
+ * @param <R> the {@link CustomResource} that drives the controller
+ */
 public interface ResourceController<R extends CustomResource> {
 
   /**
-   * The implementation should delete the associated component(s). Note that this is method is
-   * called when an object is marked for deletion. After it's executed the custom resource finalizer
-   * is automatically removed by the framework; unless the return value is {@link
-   * DeleteControl#NO_FINALIZER_REMOVAL}, which indicates that the controller has determined that
-   * the resource should not be deleted yet, in which case it is up to the controller to restore the
-   * resource's status so that it's not marked for deletion anymore.
+   * This method is called when a managed custom resource instance has been marked for deletion.
+   * After execution the custom resource finalizer is automatically removed unless the return value
+   * is {@link DeleteControl#NO_FINALIZER_REMOVAL}.
```

Comment:
As previously discussed, as a developer, I want to see that the method needs to be idempotent in the javadoc because that information cannot be communicated in any other way.